src/services/expense.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+let expenses = [];
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+const getFiltered = (userId, category, from, to) => {
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.userId === userId);
+  }
+
+  if (category) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => category.includes(expense.category));
+  }
+
+  if (from) {
+    const fromDate = new Date(from);
+
+    filteredExpenses = filteredExpenses.filter(expense => {
+      const expenseDate = new Date(expense.spentAt);
+
+      return expenseDate >= fromDate;
+    });
+  }
+
+  if (to) {
+    const toDate = new Date(to);
+
+    filteredExpenses = filteredExpenses.filter(expense => {
+      const expenseDate = new Date(expense.spentAt);
+
+      return expenseDate <= toDate;
+    });
+  }
+
+  return filteredExpenses;
+};
+
+const getById = (expenseId) => {
+  const foundExpense = expenses.find(({ id }) => id === expenseId);
+
+  return foundExpense || null;
+};
+
+const addExpense = (expenseData) => {
+  const maxId = Math.max(...expenses.map(expense => expense.id), 0);
+  const newId = maxId + 1;
```

Comment:
you can move this code to the function or use UUID, to avoid potential collisions when creating new expenses.